qemu/tests/nic_opt.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from virttest import utils_net
+from virttest import utils_test
+from virttest import error_context
+from avocado.utils import cpu
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    NIC option check test.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def test_netperf():
+        """
+        Netperf stress test for nic option.
+        """
+        netperf_client = params.get("netperf_client", "main_vm")
+        netperf_server = params.get("netperf_server", "localhost")
+        sub_type = params.get("sub_type", "netperf_stress")
+        params["netperf_client"] = netperf_client
+        params["netperf_server"] = netperf_server
+        error_context.context("Run netperf stress test, client: %s, server: %s" %
+                              (netperf_client, netperf_server), logging.info)
+        utils_test.run_virt_sub_test(test, params, env, sub_type)
+
+    def test_ping():
+        """
+        Ping test for nic option.
+        """
+        package_sizes = [int(i) for i in params.get("ping_package_sizes").split()]
+        if params["os_type"] == "windows":
+            ifname = utils_net.get_windows_nic_attribute(session=session,
+                                                         key="netenabled", value=True,
+                                                         target="netconnectionID")
+        else:
+            ifname = utils_net.get_linux_ifname(session,
+                                                vm.get_mac_address())
+        dest = utils_net.get_host_ip_address(params)
+
+        for size in package_sizes:
+            error_context.context("Test ping from '%s' to host '%s' on guest '%s'"
+                                  " with package size %d. " %
+                                  (ifname, dest, vm.name, size), logging.info)
+            status, output = utils_net.ping(dest=dest, count=10,
+                                            packetsize=size,
+                                            session=session,
+                                            timeout=30)
+            if status:
+                test.fail("%s ping %s unexpected, output %s" % (vm.name, dest, output))
+            if params["os_type"] == "windows":
+                # windows guest get loss=0 when ping failed sometime, need further check
+                loss = utils_test.get_loss_ratio(output)
+                if not loss and "TTL" in output:
```

Comment:
why not directly fail the case?

Reply:
windows guest get loss=0 when ping failed sometime,we need double conditions to further check.